napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1521 @@
+                        if self._fixed_aspect:
+                            if (new - fixed)[0] == 0:
+                                ratio = 1
+                            else:
+                                ratio = abs((new - fixed)[1]/(new - fixed)[0])
+                            if ratio > self._aspect_ratio:
+                                r = self._aspect_ratio/ratio
+                                new[1] = fixed[1]+(new[1]-fixed[1])*r
+                            else:
+                                r = ratio/self._aspect_ratio
+                                new[0] = fixed[0]+(new[0]-fixed[0])*r
+                        dist = np.dot(new-fixed, offset)/np.dot(size, offset)
+                        dist_perp = (np.dot(new-fixed, offset_perp) /
+                                     np.dot(size, offset_perp))
+                        scale = np.array([dist_perp, dist])
+                    elif np.mod(self._fixed_index, 4) == 1:
+                        # top selected
+                        fixed = self._fixed_vertex
+                        new = copy(coord)
+                        dist = np.dot(new-fixed, offset)/np.dot(size, offset)
+                        scale = np.array([1, dist])
+                    else:
+                        # side selected
+                        fixed = self._fixed_vertex
+                        new = copy(coord)
+                        dist_perp = (np.dot(new-fixed, offset_perp) /
+                                     np.dot(size, offset_perp))
+                        scale = np.array([dist_perp, 1])
+
+                    # prevent box from shrinking below a threshold size
+                    scene = self.viewer._canvas.scene
+                    transform = scene.node_transform(self._node)
+                    rescale = (transform.map([1, 1])[:2] -
+                               transform.map([0, 0])[:2])
+                    threshold = self._vertex_size*rescale.mean()/8
+                    scale[abs(scale*size) < threshold] = 1
+
+                    # check orientation of box
+                    angle = -np.arctan2(offset[0], -offset[1])
+                    if angle == 0:
+                        for index in self.selected_shapes:
+                            self.data.scale(index, scale,
+                                            center=self._fixed_vertex)
+                        self._scale_box(scale, center=self._fixed_vertex)
+                    else:
+                        rotation = np.array([[np.cos(angle), np.sin(angle)],
+                                            [-np.sin(angle), np.cos(angle)]])
+                        scale_mat = np.array([[scale[0], 0], [0, scale[1]]])
+                        inv_rot = np.array([[np.cos(angle), -np.sin(angle)],
+                                           [np.sin(angle), np.cos(angle)]])
+                        transform = np.matmul(rotation,
+                                              np.matmul(scale_mat, inv_rot))
```

Comment:
Oh this is exactly what `@` is for!!!

`transform = rotation @ scale_mat @ inv_rot`

Reply:
thanks for the tip!